lock-api-objects/src/main/java/com/palantir/lock/annotations/ReviewedRestrictedApiUsage.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.annotations;
+
+/**
+ * Annotation to be used to indicate that usage of a restricted method is intentional and all the nuances about it
+ * are understood.
+ */
+public @interface ReviewedRestrictedApiUsage {}
```

Comment:
This is a copy of a class that Lucas created when he added `getWithExpected*` API. I could not re-use his because it was located in the transaction api package.

And, moving it to some commons package is awkward. Unless I've missed something, you would need to depend on that package as "api" in Gradle (to expose it to users, e.g. internal library). But, then you pull everything from that commons package.

I could create just one package whose whole job is encapsulating this, but I figured it would be useful to have separate annotations for these. I can rename this one though if we want to disambiguate!

Reply:
I don't really like using annotations to encode stuff like this. We should instead have a more concerted effort to separate PUBLIC from PRIVATE APIs. Right now e.g. TimelockService is a hodge podge of stuff we NEVER want anyone but us to use and stuff that's maybe useful to other people, but we should probably provide a tighter API